framework/wazuh/core/batcher/mux_demux.py

```diff
@@ -0,0 +1,249 @@
+    msg : dict
+        The message content as a dictionary.
+    """
+    def __init__(self, uid: int, msg: dict):
+        self.uid = uid
+        self.msg = msg
+
+
+class MuxDemuxQueue:
+    """A queue for managing messages between mux and demux components.
+
+    Parameters
+    ----------
+    proxy_dict : DictProxy
+        A dictionary-like proxy for managing responses.
+    mux_queue : Queue
+        The queue for multiplexing messages.
+    demux_queue : Queue
+        The queue for demultiplexing messages.
+    """
+    def __init__(self, proxy_dict: DictProxy, mux_queue: Queue, demux_queue: Queue):
+        self.responses = proxy_dict
+        self.mux_queue = mux_queue
+        self.demux_queue = demux_queue
+
+    def send_to_mux(self, uid: int, msg: dict) -> int:
+        """Puts a message into the mux queue with an associated unique identifier.
+
+        Parameters
+        ----------
+        uid : int
+            The unique identifier for the message.
+        msg : dict
+            The message content to be put into the mux queue.
+
+        Returns
+        -------
+        int
+            The unique identifier of the message.
+        """
+        msg = Message(uid=uid, msg=msg)
+        self.mux_queue.put(msg)
+        return uid
+
+    def receive_from_mux(self, block: bool = True) -> Message:
+        """Retrieves a message from the mux queue. f the queue
+        is empty and block is False it raises an queue.Empty error
+
+        Returns
+        -------
+        Message
+            The message retrieved from the mux queue.
+        """
+        message = self.mux_queue.get(block=block)
+        return message
+
+    def send_to_demux(self, msg: Message):
+        """Puts a message into the demux queue.
+
+        Parameters
+        ----------
+        msg : Message
+            The message to be put into the demux queue.
+        """
+        self.demux_queue.put(msg)
+
+    def is_response_pending(self, uid: int) -> bool:
+        """Checks if a response is available for a given unique identifier.
+
+        Parameters
+        ----------
+        uid : int
+            The unique identifier to check.
+
+        Returns
+        -------
+        bool
+            True if the response is available, False otherwise.
+        """
+        return uid not in self.responses
+
+    def receive_from_demux(self, uid: int) -> Optional[dict]:
+        """Retrieves and removes a response from the dictionary for a given unique identifier.
+
+        Parameters
+        ----------
+        uid : int
+            The unique identifier for the response.
+
+        Returns
+        -------
+        Optional[dict]
+            The response dictionary if available, None otherwise.
+        """
+        if not self.is_response_pending(uid):
+            response = self.responses[uid]
+            del self.responses[uid]
+            return response
+        else:
+            return None
```

Comment:
Not really necessary

```suggestion
```

Reply:
Fixed in cc8e51ccec60b8e9576fb54270b2182bef34530f.